hal/src/nRF52840/adc_hal.cpp

```diff
@@ -38,6 +38,7 @@ static const nrfx_saadc_config_t saadcConfig = {
 
 static hal_adc_reference_t adcReference = HAL_ADC_REFERENCE_EXTERNAL;
 static constexpr uint32_t HAL_VERSION_BRN404X_V003 = 0x03;
+static constexpr uint32_t HW_FEATURE_FLAG_USE_INTERNAL_ADC_REFERENCE_BIT = (1<<15);
```

Comment:
Double check endianness.

Reply:
`hal_get_device_hw_info()` Puts the bytes in the right spots, I can examine `adcReference` with debugger after HAL init and can see its detecting the cleared bit in the right spot:
```
(gdb) p adcReference
$1 = HAL_ADC_REFERENCE_INTERNAL
```